### PR TITLE
cmus: fix feature detection on darwin

### LINKED ISSUE #271523: cmus working partially on darwin
### Describe the bug
I'm on macOS (sonoma, `14.1.1 (23B81)`) and my packages are managed by `home-manager`.

When I launch `cmus`, I get a few errors. The text is only in black and white, and I can't seem to switch between panes using `tab`.
Sadly, I didn't capture the three lines of minor errors and I'm unable to reproduce them. They seem to be related to decoders.

I'm filing this issue because cmus is known to work on darwin with the binary obtained with homebrew, so I'm suspecting something might be off in the nix package.

### Steps To Reproduce
1. `nix-shell -p cmus -c fish`

### Expected behavior
- Theming should work
- Tab should work (for pane switching)
- shouldn't show error at startup

### Screenshots
not working (nix):
image
working (brew) (please ignore the lyrics, that is a tmux pane): unknown

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Thank you for your help :)

### PR DESCRIPTION
**Pull request by emilytrau**
## Description of changes

Fixes #271523

@leana8959 would you please test if this resolves your issues? `nix run github:emilytrau/nixpkgs/cmus-darwin#cmus`

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/cmus/default.nix [hunk 1]
```diff
@@ -104,7 +104,15 @@ stdenv.mkDerivation rec {
   patches = [
     ./option-debugging.patch
     # ffmpeg 6 fix https://github.com/cmus/cmus/pull/1254/
-    (fetchpatch { url = "https://github.com/cmus/cmus/commit/07b368ff1500e1d2957cad61ced982fa10243fbc.patch"; hash = "sha256-5gsz3q8R9FPobHoLj8BQPsa9s4ULEA9w2VQR+gmpmgA="; })
+    (fetchpatch {
+      url = "https://github.com/cmus/cmus/commit/07b368ff1500e1d2957cad61ced982fa10243fbc.patch";
```
Comment (risicle) [inline pkgs/applications/audio/cmus/default.nix:108]:
```suggestion
      name = "ffmpeg-6-compat.patch";
      url = "https://github.com/cmus/cmus/commit/07b368ff1500e1d2957cad61ced982fa10243fbc.patch";
```

File: pkgs/applications/audio/cmus/default.nix [hunk 2]
```diff
@@ -104,7 +104,15 @@ stdenv.mkDerivation rec {
...
+      hash = "sha256-5gsz3q8R9FPobHoLj8BQPsa9s4ULEA9w2VQR+gmpmgA=";
+    })
+    # function detection breaks with clang 16
+    (fetchpatch {
+      url = "https://github.com/cmus/cmus/commit/4123b54bad3d8874205aad7f1885191c8e93343c.patch";
```
Comment (risicle) [inline pkgs/applications/audio/cmus/default.nix:113]:
```suggestion
      name = "clang-16-function-detection.patch";
      url = "https://github.com/cmus/cmus/commit/4123b54bad3d8874205aad7f1885191c8e93343c.patch";
```

**Comment by risicle** [comment]
Other than that, WFM macos 12 x86_64 & nixos x86_64

**Comment by leana8959** [comment]
Hello,

Are there any updates about this fix? It would be really nice to be able to use `cmus` with nix. Alternatively, is there a way I could put this branch with the fix in my `home-manager`'s configuration so that I could use the fixed `cmus` while waiting it to be merged?

Thank y'all so much again for fixing it <3 !

**Comment by emilytrau** [comment]
My apologies this slipped my mind! I've made those changes and it should be good to go now

**Comment by leana8959** [comment]
Hello,
It’s ok! Though I did notice that the ffmpeg patch has been added with #264043, maybe we need to remove that?

The overlay worked like a charm, thanks <3

**Pull request merged**